modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java

```diff
@@ -543,28 +570,10 @@ else if (Objects.equals(BinaryExpression.Operation.OR, operation)) {
 				Predicate.withParentheses((Predicate)right));
 		}
 		else if (Objects.equals(left, "keywords")) {
-			return _getColumn(
-				"id", objectDefinitionId
-			).in(
-				DSLQueryFactoryUtil.select(
-					AssetEntryTable.INSTANCE.classPK
-				).from(
-					AssetEntryTable.INSTANCE
-				).innerJoinON(
-					AssetEntries_AssetTagsTable.INSTANCE,
-					AssetEntryTable.INSTANCE.entryId.eq(
-						AssetEntries_AssetTagsTable.INSTANCE.entryId)
-				).innerJoinON(
-					AssetTagTable.INSTANCE,
-					AssetTagTable.INSTANCE.tagId.eq(
-						AssetEntries_AssetTagsTable.INSTANCE.tagId
-					).and(
-						_getExpressionPredicate(
-							AssetTagTable.INSTANCE.name, operation,
-							(String)right)
-					)
-				)
-			);
+			return _getKeywordsPredicate(
```

Comment:
Picky comment: maybe, to avoid having more returns statements you can follow the pattern followed in the method and instead of putting a return, put `predicate = _getColumn(...`

Reply:
True, thanks!